tutorials/nlp/Dialogue.ipynb

```diff
@@ -1,717 +1,2338 @@
+    "## Training and/or Testing the model\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "fwGQCwbvRf2m"
+   },
+   "outputs": [],
+   "source": [
+    "# model.dataset.data_dir: folder to load data from\n",
+    "# model.dataset.dialogues_example_dir: folder that stores predictions for each sample\n",
+    "\n",
+    "!(python NeMo/examples/nlp/dialogue/dialogue.py \\\n",
+    "  do_training=True \\\n",
+    "  model.dataset.dialogues_example_dir='./marco_bart_predictions' \\\n",
+    "  model.dataset.data_dir='./ms_marco' \\\n",
+    "  model.save_model=True \\\n",
+    "  model.dataset.debug_mode=True \\\n",
+    "  model.dataset.task='ms_marco' \\\n",
+    "  model.language_model.pretrained_model_name='facebook/bart-base' \\\n",
+    "  trainer.max_epochs=1 \\\n",
+    "  model.dataset.debug_mode=False \\\n",
+    "  exp_manager.create_wandb_logger=False)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "UL7ekAOZ2abi"
+   },
+   "source": [
+    "**After 1 epoch:**\n",
+    "\n",
+    "Train more epochs for optimal performance\n",
+    "\n",
+    "```\n",
+    "────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────\n",
+    "       Test metric             DataLoader 0\n",
+    "────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────\n",
+    "          bleu               65.46179962158203\n",
+    "           f1                78.24439835896995\n",
+    "        precision            81.92473076099847\n",
+    "         recall              76.72508929408436\n",
+    "      test_accuracy         25.563487607283225\n",
+    "        test_loss           0.4419259166606655\n",
+    "     test_loss_epoch        0.4420809745788574\n",
+    "        test_ppl            1.5557004846779854\n",
+    "────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# 4. Zero Shot Slot Filling using Assistant Dataset\n",
+    "\n",
+    "## 4.1 Task Description\n",
+    "\n",
+    "**Zero Shot Slot Filling** - is a task of detecting mention from BIO slot filling and classifying the associated slot class (Entities)\n",
+    "\n",
+    "For example, in the query:  `What is the weather in Santa Clara tomorrow morning?`, we would like to detect `Santa Clara` as a `location` slot and `tomorrow morning` as a `date_time` slot.\n",
+    "\n",
+    "Slots names are usually task specific and defined as labels in the training data.\n",
+    "This is a fundamental step that is executed in any task-driven Conversational Assistant.\n",
+    "\n",
+    "Our model enables to fine-tuned on in domain dataset and test. Also enables to transfer fine-tuned model to test out of domain dataset.\n",
```

Comment:
something like "Our model enables to fine-tune a model on in-domain dataset as well as enables to transfer fine-tuned models to test out of domain dataset."